server/src/lib/data.ts

```diff
@@ -0,0 +1,95 @@
+import {db} from "./db"
+import BlogSchema from "../Schema/Blog"
+import {z} from "zod"
+import { PrismaClient, Category } from '@prisma/client';
+
+
+// getting all the available blogs in the databse 
+export async function getBlog(){
+    return db.blog.findMany()
+}
+
+export async function getBlogByAuthorId(authorId: string) {
+    return db.blog.findMany({
+        where : {
+            authorId,
+        },
+    })
+}
+
+// Create
```

Comment:
1. Create an author schema with zod, similar to how you did with Blog schema
2. You don't need to create params, use `z.infer<typeof <youschema>>`
like this
`export async function createAuthor(data: z.infer<typeof AuthorSchema>)`
this will validate the data for you so you don't need to do it manually